entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -626,5 +626,38 @@ class EntityService(
             )
 
         throw ResourceNotFoundException("Default instance of $expandedAttributeName not found in entity $entityId")
+
     }
+
+    @Transactional(readOnly = true)
+    fun searchAuthorizedEntities(
+        queryParams: QueryParams,
+        sub: Option<Sub>,
+        offset: Int,
+        limit: Int,
+        contextLink: String,
+        includeSysAttrs: Boolean
+    ): Pair<Int, List<JsonLdEntity>> {
+        val result = searchRepository.getAuthorizedEntities(
+            queryParams,
+            sub,
+            offset,
+            limit,
+            listOf(contextLink)
+        )
+
+        return Pair(result.first, getAuthorizedEntities(result.second, includeSysAttrs))
+    }
+    //Maybe add parameter in getFullEntitiesById to have only type and id in JsonLdEntity
+    fun getAuthorizedEntities(entitiesIds: List<Entity>, includeSysAttrs: Boolean = false): List<JsonLdEntity> =
+        entitiesIds
+            .map {
+                JsonLdEntity(
+                    it.serializeCoreProperties(includeSysAttrs),
+                    it.contexts
+                )
+            }.sortedBy {
```

Comment:
not needed anymore?

Reply:
This allows me to have `createdAt` and `modifiedAt`.

Reply:
`sortBy` yes but `it.serializeCoreProperties(includeSysAttrs)` it's use for `createdAt` and `modifiedAt`

Reply:
yes, I was referring to `sortedBy`